core/src/main/java/org/carbondata/core/cache/dictionary/AbstractColumnDictionaryInfo.java

```diff
@@ -241,22 +257,21 @@ private void decrementAccessCount() {
    */
   protected byte[] getDictionaryBytesFromSurrogate(int surrogateKey) {
     byte[] dictionaryValueInBytes = null;
-    int totalSizeOfDictionaryChunksTraversed = 0;
-    for (List<byte[]> oneDictionaryChunk : dictionaryChunks) {
-      totalSizeOfDictionaryChunksTraversed =
-          totalSizeOfDictionaryChunksTraversed + oneDictionaryChunk.size();
-      // skip the dictionary chunk till surrogate key is lesser than size of
-      // dictionary chunks traversed
-      if (totalSizeOfDictionaryChunksTraversed < surrogateKey) {
-        continue;
+    if (surrogateKey > 0) {
```

Comment:
why we need this check ? As data loading surrogate key starts from 1 , so in any case if surrogate key is less than 1 then  there is some bug in unpacking the mdkey